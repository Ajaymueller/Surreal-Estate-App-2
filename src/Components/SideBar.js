import React, { useState } from "react";
import { Link, useLocation, useHistory } from "react-router-dom";
import { buildQueryString } from "../helpers";
import SideBarForm from './SideBarForm';

import "../styles/SideBar.css";

const cities = ["Manchester", "Leeds", "Sheffield", "Liverpool"];

const SideBar = () => {
  const { search } = useLocation();
  const { push } = useHistory();
  const [searchValue, setSearchValue] = useState("");

  const handleSearch = (event) => {
    event.preventDefault();
    const newQueryString = buildQueryString(
      "query",
      { title: { $regex: searchValue } },
      search
    );

    push(newQueryString);
  };

  const handleChange = (e) => {
    setSearchValue(e.target.value);
};

  return (
    <div className="SideBar">
     <SideBarForm 
            handleChange={handleChange}
            handleSearch={handleSearch}
            query={searchValue}/>
      <div>
        <h3>Filter By</h3>
        {cities.map((city) => (
          <Link key={city} to={buildQueryString("query", { city }, search)}>
            <li data-testid="city-option">{city}</li>
          </Link>
        ))}
      </div>
      <div>
        <h3>Sort By</h3>
        <Link to={buildQueryString("sort", { price: +1 }, search)}>
          <li>Ascending</li>
        </Link>
        <Link to={buildQueryString("sort", { price: -1 }, search)}>
          <li>Descending</li>
        </Link>
      </div>
    </div>
  );
};

export default SideBar;