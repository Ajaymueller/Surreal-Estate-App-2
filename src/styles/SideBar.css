.SideBar {
    width: 180px; 
    position: sticky;
    margin-top: 44px;
    left: 0;
    top: 0; 
    display: flex;
    flex-wrap: wrap;
    flex-direction: column;
    padding: 5px;
   /* z-index: 1; 
    top: 0; 
    left: 0;
    margin-top: 44px;
    overflow-x: hidden; 
    padding-top: 20px;
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    padding: 5px;
    height: 100vh;*/
}

.sideBar-form {
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    width: 100%;
}

.SideBar a,
h3 {
  padding: 6px 8px 6px 16px;
  text-decoration: none;
  font-size: 25px;
  color: #818181;
  display: block;
}

.SideBar a:hover {
  color: #f1f1f1;
}

.SideBar button {
    margin-top: 8px;
    background-color: #4CAF50;
    border-radius: 10px;
}

.SideBar button:hover {
    color:#BBBBBB;
    border-color:#BBBBBB;
    cursor: pointer;
}

.SideBar li {
    margin: 10px;
    list-style: none;
    padding: 12px 10px;
    border-bottom: 2px;
}

.SideBar h3 {
  padding: 12px 10px;
  border-bottom: 2px;
}