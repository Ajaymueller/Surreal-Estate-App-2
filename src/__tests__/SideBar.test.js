import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import SideBar from '../Components/SideBar';
import { MemoryRouter } from 'react-router-dom';
import {withRouter} from 'react-router'
import '@testing-library/jest-dom/extend-expect';

describe("SideBar", () => {
    it("render correctly", () => {
    const { asFragment } = render(
    <MemoryRouter><SideBar /></MemoryRouter>)
    expect(asFragment).toMatchSnapshot(); 
    });
    it("sidebar links have option to filter by and sort by", () => {
    const { getByText } = render(
    <MemoryRouter><SideBar /></MemoryRouter>)
    const filterBy= getByText("Filter By")
    const sortBy = getByText("Sort By")
    expect(filterBy).toBeInTheDocument();
    expect(sortBy).toBeInTheDocument();
    });
    it("has li that displays 4 different city options", () => {
    const { getAllByTestId } = render(
    <MemoryRouter><SideBar /></MemoryRouter>)
    const city = getAllByTestId("city-option")
    expect(city).toHaveLength(4);
    });
    it("can sort options by ascending or descending", () => {
    const { getByText } = render(
    <MemoryRouter><SideBar /></MemoryRouter>)
    const ascending = getByText("Ascending");
    const descending = getByText("Descending");
    expect(ascending).toBeInTheDocument();
    expect(descending).toBeInTheDocument();
    })
    it("contains 6 link elements", () => {
    const { getAllByRole } = render(
    <MemoryRouter><SideBar /></MemoryRouter>)
    const links = getAllByRole("link");
    expect(links).toHaveLength(6);
    });
});

