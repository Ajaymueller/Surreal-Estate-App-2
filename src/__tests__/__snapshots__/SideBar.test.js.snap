exports[`SideBar render correctly 1`] = `[Function]`;
